api_tests/draft_registrations/views/test_draft_registration_list.py

```diff
@@ -105,39 +108,55 @@ def payload(self, metaschema_open_ended, provider, project_public):
             }
         }
 
+    # Temporary alternative provider that supports `metaschema_open_ended` in `TestDraftRegistrationCreate`
+    # This provider is created to fix the first 3 tests in this test class due to test DB changes with the
+    # Django 3 Upgrade. A long-term solution is to create and/or use dedicated schemas for testing.
+    @pytest.fixture()
+    def provider_alt(self, metaschema_open_ended):
+        default_provider = RegistrationProvider.get_default()
+        default_provider.schemas.add(metaschema_open_ended)
+        default_provider.save()
+        return default_provider
```

Comment:
`TestDraftRegistrationCreateWithNode` inherits from `TestDraftRegistrationCreate`. Thus modifying the provider in the latter breaks `TestDraftRegistrationCreate`. In addition, some tests  in `TestDraftRegistrationCreateWithNode` also relied on this version of `metaschema_open_ended` not in provider.

This issue is caused by our factories relies on test DB (i.e. adding the `pk=1` schema when calling `RegistrationFactory`) instead of adding a schema with specified name and pinned version. Pre and post upgrade test DB are slightly different which breaks the tests.